### PR TITLE
Fixed some Apple platform libs

### PR DESCRIPTION
**Pull request by homuroll**

### CODE REVIEW

File: platformLibs/src/platform/ios/UIKit.def [hunk 1]
```diff
@@ -5,3 +5,4 @@ modules = UIKit
 
 compilerOpts = -framework UIKit
 linkerOpts = -framework UIKit
+excludedFunctions = UISceneErrorDomain
```
Comment (SvyatoslavScherbina) [inline platformLibs/src/platform/ios/UIKit.def]: I can't find this function in `UIKIt`. Where is it located?

File: platformLibs/src/platform/ios/builtin.def [hunk 2]
```diff
@@ -2,6 +2,9 @@ depends = posix
 package = platform.builtin
 headerFilter = **
 language = C
+
+excludedFunctions.ios_arm32 = builtin_clear_cache
+excludedFunctions.ios_arm64 = builtin_clear_cache
```
Comment (SvyatoslavScherbina) [inline platformLibs/src/platform/ios/builtin.def]: What are these ones?
Which binary contains definitions of these functions on ios_x64?

File: platformLibs/src/platform/ios/UIKit.def (hunk 1 above)
Comment (SvyatoslavScherbina) [reply platformLibs/src/platform/ios/UIKit.def]: I see, `excludedFunctions` filters globals too.

File: platformLibs/src/platform/ios/builtin.def (hunk 2 above)
Comment (homuroll) [reply platformLibs/src/platform/ios/builtin.def]: This function is defined below in the same def file. It calls `__builtin___clear_cache`, which is for some reason missing on ios_armXX.

File: platformLibs/src/platform/ios/UIKit.def (hunk 1 above)
Comment (homuroll) [reply platformLibs/src/platform/ios/UIKit.def]: `UIKit.framework/Headers/UISceneDefinitions.h`

File: platformLibs/src/platform/osx/darwin.def [hunk 3]
```diff
@@ -33,4 +33,4 @@ headerFilter = **
 excludedFunctions = __tg_promote
 
 compilerOpts = -D_XOPEN_SOURCE -DSHARED_LIBBIND -D_DARWIN_NO_64_BIT_INODE -DSYSCTL_DEF_ENABLED
-linkerOpts = -ldl -lz
+linkerOpts = -ldl -lz -lcurses -undefined dynamic_lookup
```
Comment (SvyatoslavScherbina) [inline platformLibs/src/platform/osx/darwin.def]: This affects linkage of all libraries and Kotlin code.
I believe we should find better solution if possible.
Would supporting regexp in `excludedFunctions` help?

File: platformLibs/src/platform/ios/darwin.def [hunk 4]
```diff
@@ -29,10 +29,15 @@ headers = AppleTextureEncoder.h AssertMacros.h Availability.h AvailabilityIntern
...
+excludedFunctions = __tg_promote KERNEL_AUDIT_TOKEN KERNEL_SECURITY_TOKEN \
+                    asl_decode_buffer asl_encode_buffer asl_remove_index \
+                    averunnable dbm_forder pidlock ttyaction ttylock ttymsg ttyunlock uuid_generate_early_random
+
+excludedFunctions.ios_arm32 = _Unwind_Backtrace _Unwind_FindEnclosingFunction _Unwind_Find_FDE \
+                              _Unwind_SjLj_ForcedUnwind __deregister_frame __register_frame
 
 compilerOpts = -D_XOPEN_SOURCE -DSYSCTL_DEF_ENABLED
-linkerOpts = -ldl -lz
+linkerOpts = -ldl -lz -lbz2 -lcompression -late
```
Comment (SvyatoslavScherbina) [inline platformLibs/src/platform/ios/darwin.def]: Isn't the same required for macOS?

File: platformLibs/src/platform/osx/darwin.def (hunk 3 above)
Comment (olonho) [reply platformLibs/src/platform/osx/darwin.def]: Yeah, but maybe not in this PR, but in the followup where we'll use static libs.

File: platformLibs/src/platform/ios/darwin.def (hunk 4 above)
Comment (homuroll) [reply platformLibs/src/platform/ios/darwin.def]: Indeed.

File: platformLibs/src/platform/osx/darwin.def (hunk 3 above)
Comment (homuroll) [reply platformLibs/src/platform/osx/darwin.def]: Actually, after adding `bz2`, `compression`, `ate` and `bsm` libs only a couple of dozens of functions remain unresolved and they can be explicitly added to `excludedFunctions`.

File: platformLibs/src/platform/ios/CoreAudioTypes.def [hunk 5]
```diff
@@ -4,4 +4,3 @@ package = platform.CoreAudioTypes
 modules = CoreAudioTypes
 
 compilerOpts = -framework CoreAudioTypes
-linkerOpts = -framework CoreAudioTypes
```
Comment (SvyatoslavScherbina) [inline platformLibs/src/platform/ios/CoreAudioTypes.def]: + watchOS, tvOS

**Pull request merged**